bentoml/_internal/runner/runnable.py

```diff
@@ -13,60 +14,74 @@
 
 import attr
 
+from bentoml._internal.types import AnyType
 from bentoml._internal.types import LazyType
 
 logger = logging.getLogger(__name__)
 
 RUNNABLE_METHOD_MARK: str = "_bentoml_runnable_method"
 
 
-def method_decorator(
-    meth: WrappedMethod,
-    batchable: bool = False,
-    batch_dim: BatchDimType = 0,
-    input_spec: LazyType[t.Any] | t.Tuple[LazyType[t.Any], ...] | None = None,
-    output_spec: LazyType[t.Any] | None = None,
-) -> WrappedMethod:
-    setattr(
-        meth,
-        RUNNABLE_METHOD_MARK,
-        RunnableMethodConfig(
-            batchable=batchable,
-            batch_dim=batch_dim,
-            input_spec=input_spec,
-            output_spec=output_spec,
-        ),
-    )
-    return meth
+class Runnable(ABC):
+    @property
+    @abstractmethod
+    def supports_nvidia_gpu(self) -> bool:
```

Comment:
This can not be a property because we need this before init the runnable.

Reply:
Ah, I meant for these to be static but apparently that's not possible, will revert.